compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2522,7 +2526,10 @@ fn add_native_libs_from_crate(
             NativeLibKind::WasmImportModule => {}
             NativeLibKind::LinkArg => {
                 if link_static {
-                    cmd.arg(name);
+                    let need_wrap = flavor.uses_cc() && !verbatim;
+                    let wrapped_arg =
+                        if need_wrap { "-Wl,".to_owned() + name } else { name.to_string() };
+                    cmd.arg(wrapped_arg);
```

Comment:
Linker flavor is already baked into `cmd`, so I suggest adding a new `fn link_arg(verbatim: bool)` method to `trait Linker` (perhaps with a default implementation).

`uses_cc` also returns true for `LinkerFlavor::EmCc`, which has quite special interface, but maybe let's also add `-Wl,` for it too, because it's a clang+lld in disguise.

Reply:
Implemented linker_arg/linker_args Linker trait methods and used for link_arg processing.